### PR TITLE
Add metrics package and support for mockery

### PR DESCRIPTION
**Pull request by nashluffy**
Primary reason for pulling this out to it's own package is so we can mock the metrics calls in unit tests.

### CODE REVIEW

File: internal/metrics/metrics.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
+//go:generate mockery --name Client
+package metrics
+
+import (
```
Comment (toastwaffle) [inline internal/metrics/metrics.go:4]: group imports?

File: internal/metrics/metrics.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+type Client interface {
+	IncreaseInterruptionEventCounter(cluster string)
+	RegisterHTTPHandler()
+}
+
+func (m *metrics) IncreaseInterruptionEventCounter(cluster string) {
+	interruptionEvents.WithLabelValues(cluster).Inc()
+}
+
+func (m *metrics) RegisterHTTPHandler() {
```
Comment (toastwaffle) [inline internal/metrics/metrics.go]: This should probably be something like `ServeMetrics`

File: internal/metrics/metrics.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+)
+
+var (
+	interruptionEvents = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "interruption_events_total",
+		Help: "The total number of spot interruptions for a given cluster",
+	}, []string{"kubernetes_cluster"})
+)
+
+type Client interface {
```
Comment (toastwaffle) [inline internal/metrics/metrics.go:22]: I don't think `Client` is the right name for this - it's not making any calls to a server. I think just call it `Metrics`?

Comment (toastwaffle) [inline internal/metrics/metrics.go:22]: Add doc comments

**Review by toastwaffle — CHANGES_REQUESTED** [review]

File: internal/metrics/metrics.go (hunk 1 above)
Comment (nashluffy) [reply internal/metrics/metrics.go:4]: will handle with linter and precommit after these changes have been landed

File: internal/metrics/metrics.go (hunk 3 above)
Comment (nashluffy) [reply internal/metrics/metrics.go:22]: metrics.Metrics is bad too. I didn't mean client in the sense of client-server, but more in the sense of "use this client that interfaces with the underlying metrics implementation". I can't think of anything more appropriate but open to more suggestions!

**Review by toastwaffle — APPROVED** [review]

**Pull request merged**